src/main/java/uk/gov/hmcts/reform/prl/services/HwfProcessUpdateCaseStateService.java

```diff
@@ -0,0 +1,190 @@
+import uk.gov.hmcts.reform.prl.models.dto.ccd.request.Match;
+import uk.gov.hmcts.reform.prl.models.dto.ccd.request.Must;
+import uk.gov.hmcts.reform.prl.models.dto.ccd.request.Query;
+import uk.gov.hmcts.reform.prl.models.dto.ccd.request.QueryParam;
+import uk.gov.hmcts.reform.prl.models.dto.ccd.request.Range;
+import uk.gov.hmcts.reform.prl.models.dto.ccd.request.Should;
+import uk.gov.hmcts.reform.prl.models.dto.ccd.request.StateFilter;
+import uk.gov.hmcts.reform.prl.models.dto.payment.PaymentGroupReferenceStatusResponse;
+import uk.gov.hmcts.reform.prl.services.tab.alltabs.AllTabServiceImpl;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import static org.apache.commons.collections.CollectionUtils.isNotEmpty;
+import static uk.gov.hmcts.reform.prl.constants.PrlAppsConstants.CASE_TYPE;
+import static uk.gov.hmcts.reform.prl.enums.CaseEvent.HWF_PROCESS_CASE_UPDATE;
+
+@Slf4j
+@Service
+@Component
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class HwfProcessUpdateCaseStateService {
+
+    private final SystemUserService systemUserService;
+    private final AuthTokenGenerator authTokenGenerator;
+    private final CoreCaseDataApi coreCaseDataApi;
+    private final PaymentRequestService paymentRequestService;
+    private final AllTabServiceImpl allTabService;
+
+    private final ObjectMapper objectMapper;
+
+
+    public void checkHwfPaymentStatusAndUpdateCaseState() {
+        long startTime = System.currentTimeMillis();
+        log.info("inside checkHwfPaymentStatus");
+        //Fetch all C100 pending cases with Help with fees
+        List<CaseDetails> caseDetailsList = retrieveCasesWithHelpWithFeesInPendingState();
+        if (isNotEmpty(caseDetailsList)) {
+            caseDetailsList.forEach(caseDetails -> {
+                log.info("caseDetails caseId - " + caseDetails.getId());
+                CaseData caseData = objectMapper.convertValue(caseDetails.getData(), CaseData.class);
+                if (StringUtils.isNotEmpty(caseData.getHelpWithFeesNumber())
+                    && StringUtils.isNotEmpty(caseData.getPaymentServiceRequestReferenceNumber())) {
+                    log.info("Going to check service request payment status");
+                    PaymentGroupReferenceStatusResponse paymentGroupReferenceStatusResponse = paymentRequestService.fetchPaymentGroupReferenceStatus(
+                        systemUserService.getSysUserToken(),
+                        caseData.getPaymentServiceRequestReferenceNumber()
+                    );
+                    log.info("PaymentGroupReferenceStatusResponse - " + paymentGroupReferenceStatusResponse.getServiceRequestStatus());
+                    if (!PaymentStatus.PAID.getDisplayedValue().equals(paymentGroupReferenceStatusResponse.getServiceRequestStatus())) {
```

Comment:
@sahaSonali Please correct it & also check if we need anything to handle partial remissions.